products/memgraph/v0.51.0/reference_guide/query-modules.md

```diff
@@ -196,6 +200,123 @@ As previously mentioned, no exceptions should leave your module. If you are
+
+- `args`, a copy of arguments passed to the procedure.
+- `vertex_count`, number of vertices in the database.
+- `avg_degree`, average degree of vertices.
+- `props`, properties map of the Vertex or Edge object passed in `required_arg`.
+   In case a Path instance is passed, procedure returns the properties map of
+   the starting vertex.
+
+This procedure can be invoked in openCypher as follows:
+
+```opencypher
+MATCH (n) CALL py_example.procedure(n, 1) YIELD * RETURN *;
+```
+
+The following lines create the properties map for a received Edge, Vertex
+or Path instance:
+
+```python
+if isinstance(required_arg, (mgp.Edge, mgp.Vertex)):
+    props = dict(required_arg.properties.items())
+elif isinstance(required_arg, mgp.Path):
+    start_vertex, = required_arg.vertices
+    props = dict(start_vertex.properties.items())
+```
+
+As you can see, in the case of `mgp.Edge` and `mgp.Vertex`, we obtain an
+instance of `mgp.Properties` class which holds the respective properties by
+accessing the `properties` property of our `mgp.Edge` or `mgp.Vertex` instance.
+Once we have access to `mgp.Properties` instance, we can simply invoke the
+`items()` method which returns an `Iterable` that contains `mgp.Property`
+objects. Since the type of `mgp.Property` is a simple `collections.namedtuple`
+containing `name` and `value`, we can easily pass it to a `dict` constructor.
+
+We go on to counting the number of vertices and edges in our graph:
+
+```python
+vertex_count = 0
+edge_count = 0
+for v in context.graph.vertices:
+    vertex_count += 1
+    edge_count += sum(1 for e in v.in_edges)
+    edge_count += sum(1 for e in v.out_edges)
+```
+
+As you can see, we can access the `mgp.Graph` instance through `context.graph`.
+This instance contains the state of our database when executing the cypher query
+that called our procedure. A `mgp.Graph` instance has a property `vertices`
+which allows us to access a `mgp.Vertices` object which can be iterated upon.
+
+Similarly, each `mgp.Vertex` object has `in_edges` and `out_edges` properties
+which allow us to iterate over corresponding `mgp.Edge` objects. The rest of the
+code logic from the previous snippet is self-explanatory, we simply increase the
+adequate variables on each traversed vertex or edge.
+
+After that we calculate the average degree and obtain a copy of the passed
+arguments:
+
+```python
+avg_degree = 0 if vertex_count == 0 else edge_count / vertex_count
+args_copy = [copy.deepcopy(required_arg), copy.deepcopy(optional_arg)]
+```
+
+Finally, we return a `mgp.Record` with all the calculated values:
+
+```python
+return mgp.Record(args=args_copy, vertex_count=vertex_count,
+                  avg_degree=avg_degree, props=props)
+```
+
+In conclusion, Python API provided by Memgraph can be a very powerful, yet
+simple tool when implementing query modules. Therefore, we strongly suggest
+that all users thoroughly inspect the `mgp.py` source file.
+
```

Comment:
I would add a `WARNING` or `NOTE` here and tell the users that they must not use any global variables in their Python modules.